### PR TITLE
fix(pgw): updates for ipam

### PR DESCRIPTION
**Pull request by RoRoJ**
Update doc to describe legacy vs ipam gateways.

Another PR will be opened in the future once the tags of legacy or ipam are available in the console.

### CODE REVIEW

File: network/public-gateways/concepts.mdx [hunk 1]
```diff
@@ -60,6 +60,40 @@ An **I**nternet **P**rotocol address is a unique address that identifies a devic
 
 <Concept>
 
+## IPAM
+
+IPAM is Scaleway's **IP** **A**ddress **M**angement tool. Read more about it in our [VPC documentation](/network/vpc/concepts/#ipam).
```
Comment (fenollp) [inline network/public-gateways/concepts.mdx:65]:
```suggestion
IPAM is Scaleway's **IP** **A**ddress **M**anagement tool. Read more about it in our [VPC documentation](/network/vpc/concepts/#ipam).
```

File: network/public-gateways/concepts.mdx [hunk 2]
```diff
@@ -60,6 +60,40 @@ An **I**nternet **P**rotocol address is a unique address that identifies a devic
...
+
+**Legacy** Public Gateways use a [workaround](/network/vpc/reference-content/vpc-migration/#public-gateways-and-vpc) to ensure IPAM compatibility. Your gateway is a legacy gateway if:
+- You created it via the Scaleway console prior to 17 October 2023
+- You created it via the Scaleway API or devtools prior to 17 October 2023, and you did not use the `ipam_config` object when creating the [GatewayNetwork](https://www.scaleway.com/en/developers/api/public-gateway/#path-gateway-networks-attach-a-public-gateway-to-a-private-network) (attachment to a Private Network). The auto-calculated `is_legacy` [Gateway parameter](https://www.scaleway.com/en/developers/api/public-gateway/#path-gateways-create-a-public-gateway) will have a value of `true`.
```
Comment (fenollp) [inline network/public-gateways/concepts.mdx:69]:
```suggestion
- You created it via the Scaleway API or devtools prior to 17 October 2023, and you did not use the `ipam_config` object when creating the [GatewayNetwork](https://www.scaleway.com/en/developers/api/public-gateway/#path-gateway-networks-attach-a-public-gateway-to-a-private-network) (attachment to a Private Network).

The auto-calculated `is_legacy` [Gateway parameter](https://www.scaleway.com/en/developers/api/public-gateway/#path-gateways-create-a-public-gateway) will have a value of `true`. 
```

File: faq/public-gateways.mdx [hunk 3]
```diff
@@ -37,18 +37,21 @@ Moreover, the Public Gateway supports static NAT (aka. port forwarding), so that
...
 
 </Concept>
 
 <Concept>
 
-## What happened to static leases (DHCP reservations) when DHCP moved from the Public Gateway to Private Networks?
+### Why is my Public Gateway labelled as Legacy?
 
-On 12 July 2023, DHCP functionality moved from Public Gateways to Private Networks. See our [dedicated migration documentation](/network/vpc/reference-content/vpc-migration/) for full details.
+**Legacy** Public Gateways use a [workaround](/network/vpc/reference-content/vpc-migration/#public-gateways-and-vpc) to ensure compatiblity with Scaleway's IPAM (**I** **P** **A**ddress **M**anagement) tool. IPAM  IPAM acts as a single source of truth for the IP addresses of Scaleway resources
```
Comment (SamyOubouaziz) [inline faq/public-gateways.mdx:52]:
```suggestion
**Legacy** Public Gateways use a [workaround](/network/vpc/reference-content/vpc-migration/#public-gateways-and-vpc) to ensure compatiblity with Scaleway's IPAM (**I** **P** **A**ddress **M**anagement) tool. IPAM  acts as a single source of truth for the IP addresses of Scaleway resources
```

Comment (ldecarvalho-doc) [inline faq/public-gateways.mdx:52]:
```suggestion
**Legacy** Public Gateways use a [workaround](/network/vpc/reference-content/vpc-migration/#public-gateways-and-vpc) to ensure compatiblity with Scaleway's IPAM (**I** **P** **A**ddress **M**anagement) tool. IPAM acts as a single source of truth for the IP addresses of Scaleway resources
```

File: network/vpc/concepts.mdx [hunk 4]
```diff
@@ -68,7 +68,7 @@ Private Networks benefit from managed DNS, which resolves the hostnames of attac
...
+**IP** **A**ddress **M**anagement (IPAM) is a method for planning, tracking and managing the IP address space of a network. Scaleway's IPAM acts as a single source of truth for the IP addresses of Scaleway resources. Managed products, such as Databases and Load Balancer, are fully integrated into IPAM, and Private Networks' [DHCP server](#dhcp) uses IPAM to assign static IP addresses to attached resources such as Instances and Elastic Metal server. We are starting to make some public IPAM functionality available on our IPAM API: [check out the documentation](IP reservations are also available through [the IPAM API](https://www.scaleway.com/en/developers/api/ipam/) for more details.
```
Comment (SamyOubouaziz) [inline network/vpc/concepts.mdx:71]: missing link on `[check out the documentation]` ? and missing parenthesis before "for more details" ?

Comment (ldecarvalho-doc) [inline network/vpc/concepts.mdx:71]:
```suggestion
**IP** **A**ddress **M**anagement (IPAM) is a method for planning, tracking and managing the IP address space of a network. Scaleway's IPAM acts as a single source of truth for the IP addresses of Scaleway resources. Managed products, such as Databases and Load Balancer, are fully integrated into IPAM, and Private Networks' [DHCP server](#dhcp) uses IPAM to assign static IP addresses to attached resources such as Instances and Elastic Metal server. We are starting to make some public IPAM functionalities available on our IPAM API: [check out the documentation](IP reservations are also available through [the IPAM API](https://www.scaleway.com/en/developers/api/ipam/) for more details.
```

File: network/vpc/reference-content/vpc-migration.mdx [hunk 5]
```diff
@@ -62,12 +62,13 @@ From now on, **whenever you create a new Private Network**, as well as being reg
...
-- These IP addresses are managed by our internal IPAM, which is the only source of truth and ensures full consistency.
-- IP addresses are allocated and assigned to your resources: they are fixed and (for now) cannot be defined by the user, beyond definition of the CIDR block at the time of network creation. Watch this space for future developments in terms of Scaleway's IPAM and the functionality available to users.
+- These IP addresses are managed by Scaleway IPAM, which is the only source of truth and ensures full consistency.
+- IP addresses are allocated and assigned to your resources: they are fixed and (for now) cannot be defined by the user, beyond definition of the CIDR block at the time of network creation. 
+- [Scaleway's public IPAM API](https://www.scaleway.com/en/developers/api/ipam/) is now publicly available for features such as listing resource IP addresses, and booking IP addresses. As we develop our IPAM tool and integrate it with the Scaleway console, more functionality will become available such as allocating booked IPs to a resource when you attach it to a Private Network.
```
Comment (ldecarvalho-doc) [inline network/vpc/reference-content/vpc-migration.mdx:67]:
```suggestion
- [Scaleway's public IPAM API](https://www.scaleway.com/en/developers/api/ipam/) is now publicly available for features such as listing resource IP addresses, and booking IP addresses. As we develop our IPAM tool and integrate it with the Scaleway console, more functionalities will become available such as allocating booked IPs to a resource when you attach it to a Private Network.
```

File: network/vpc/concepts.mdx (hunk 4 above)
Comment (RoRoJ) [inline network/vpc/concepts.mdx:71]:
```suggestion
**IP** **A**ddress **M**anagement (IPAM) is a method for planning, tracking and managing the IP address space of a network. Scaleway's IPAM acts as a single source of truth for the IP addresses of Scaleway resources. Managed products, such as Databases and Load Balancer, are fully integrated into IPAM, and Private Networks' [DHCP server](#dhcp) uses IPAM to assign static IP addresses to attached resources such as Instances and Elastic Metal server. We are starting to make some public IPAM functionality available on our IPAM API: [check out the documentation](https://www.scaleway.com/en/developers/api/ipam/) for more details.
```

**Review by SamyOubouaziz — APPROVED** [review]

**Pull request merged**